### PR TITLE
backport: Merge bitcoin#20408,19836,19851,(partial)21424, 21948, 20056,20432,19337,20448, 21010

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 , please review

File: src/validation.cpp [hunk 1]
```diff
@@ -2771,14 +2771,6 @@ void CChainState::UpdateTip(const CBlockIndex* pindexNew)
...
             }
         }
-        // Check the version of the last 100 blocks to see if we need to upgrade:
-        for (int i = 0; i < 100 && pindex != nullptr; i++)
-        {
-            int32_t nExpectedVersion = ComputeBlockVersion(pindex->pprev, m_params.GetConsensus());
-            if (pindex->nVersion > VERSIONBITS_LAST_OLD_BLOCK_VERSION && (pindex->nVersion & ~nExpectedVersion) != 0)
-                ++nUpgraded;
-            pindex = pindex->pprev;
-        }
```
Comment (UdjinM6) [inline src/validation.cpp:2781]: 20054: looks incomplete/missing some previous backport

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/net_processing.cpp [hunk 2]
```diff
@@ -579,13 +579,13 @@ struct CNodeState {
...
+        int64_t m_timeout{0};
         //! A header with the work we require on our peer's chain
-        const CBlockIndex * m_work_header;
+        const CBlockIndex* m_work_header{nullptr};
         //! After timeout is reached, set to true after sending getheaders
-        bool m_sent_getheaders;
+        bool m_sent_getheaders{false};
         //! Whether this peer is protected from disconnection due to a bad/slow chain
-        bool m_protect;
+        bool m_protect{false};
```
Comment (knst) [inline src/net_processing.cpp:588]: `#21424` is partial because some changes can be applied only when #19109 is done.

**Review by knst — CHANGES_REQUESTED** [review]
see comment about #21424 - as partial + remove 20385 from PR title.

File: src/validation.cpp (hunk 1 above)
Comment (vijaydasmp) [reply src/validation.cpp]: removed 20054 for now , will check for dependency and then backport it

**Review by knst — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , please review

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**